homeassistant/components/dsmr/sensor.py

```diff
@@ -54,7 +55,7 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the DSMR sensor."""
     # Suppress logging
-    logging.getLogger("dsmr_parser").setLevel(logging.ERROR)
+    set_default_log_level(hass, "dsmr_parser", logging.ERROR)
```

Comment:
I wonder if this is the right approach. The problem that we're going to encounter is that we're going to encounter is that now we need to teach integrations not to use the default built-in things.

https://github.com/python/cpython/blob/master/Lib/logging/__init__.py#L1251

Looks like `logging.setLoggerClass` is a thing. We could make a small wrapper class that will change the default behavior of `setLevel` to call `set_default_log_level` and expose original `setLevel` to be used by the `logging integration.

Reply:
Sounds like a good idea. That would take care 3rdparty libs that much with the levels as well